File: BestTimetoBuyandSellStock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ll =prices[0];
        int profit=0;
        for(int i=1;i<prices.size();i++){
            int x=0;
            if(ll<prices[i]){
                x =  prices[i] - ll;
            }
            cout<<ll<<"  "<<prices[i]<<endl;
            if(profit<x){
                cout<<profit<<endl;
                profit=x;
            }
            if(ll>prices[i]){
                ll = prices[i];
            } 
        }
        return profit;
    }
};

File: Strivers SDE Sheet/Best Time to Buy and Sell Stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int mn=prices[0],mx=prices[0],ans =0;
        for(int i=1;i<prices.length;i++){
            if(mn>prices[i]){
                mn = prices[i];
                mx = prices[i];
            } else if(mx<prices[i]){
                mx = prices[i];
                if(ans<(mx-mn)){
                    ans = (mx - mn);
                }
            }
        }
        return ans;
    }
}